engine/src/main/java/org/hibernate/search/engine/backend/orchestration/spi/BatchingExecutor.java

```diff
@@ -28,16 +29,17 @@
  * Useful when works can be merged together for optimization purposes (bulking in Elasticsearch),
  * or when they should never be executed in parallel (writes to a Lucene index).
  */
-public final class BatchingExecutor<P extends BatchedWorkProcessor> {
+public final class BatchingExecutor<P extends BatchedWorkProcessor, W extends BatchedWork<? super P>> {
```

Comment:
Did you *have* to do this? Because I don't see what this brings, other than an extra generic parameter (which, by the way, is a backwards-incompatible change).

Can you revert the commit that did this, and if there's a problem with that, let's discuss it on Zulip?

Reply:
We ended up keeping the change; see https://hibernate.zulipchat.com/#narrow/stream/132092-hibernate-search-dev/topic/HSEARCH-4736